core/src/main/java/org/apache/carbondata/core/scan/result/iterator/AbstractDetailQueryResultIterator.java

```diff
@@ -126,6 +144,82 @@ private void intialiseInfos() {
     }
   }
 
+  /**
+   * Below method will be used to get the delete delta rows for a block
+   *
+   * @param dataBlock       data block
+   * @param deleteDeltaInfo delete delta info
+   * @return blockid+pageid to deleted row mapping
+   */
+  private Map<String, DeleteDeltaVo> getDeleteDeltaDetails(AbstractIndex dataBlock,
+      DeleteDeltaInfo deleteDeltaInfo) {
+    // if datablock deleted delta timestamp is more then the current delete delta files timestamp
+    // then return the current deleted rows
+    if (dataBlock.getDeleteDeltaTimestamp() >= deleteDeltaInfo
+        .getLatestDeleteDeltaFileTimestamp()) {
+      return dataBlock.getDeletedRowsMap();
+    }
+    CarbonDeleteFilesDataReader carbonDeleteDeltaFileReader = null;
+    // get the lock object so in case of concurrent query only one task will read the delete delta
+    // files other tasks will wait
+    Object lockObject = deleteDeltaToLockObjectMap.get(deleteDeltaInfo);
+    // if lock object is null then add a lock object
+    if (null == lockObject) {
+      synchronized (deleteDeltaToLockObjectMap) {
+        // double checking
```

Comment:
Again do `deleteDeltaToLockObjectMap.get(deleteDeltaInfo);` to avoid null pointer exception